tests/cases/conformance/types/specifyingTypes/typeQueries/typeofClassExpression1.ts

```diff
@@ -0,0 +1,121 @@
+// @strict: true
+// @declaration: true
+
+type TC1 = typeof class {
+    constructor(s: string);
+    static n: number;
+    s: string;
+}
+
+declare let C1: TC1;
+C1.n;
+let c1 = new C1('hello');
+c1.s;
+
+declare let C2: typeof class {
+    constructor(s: string);
+    static n: number;
+    s: string;
+}
+C2.n;
+let c2 = new C2('hello');
+c2.s;
+
+declare let C3: typeof class extends Error {
+    constructor(s: string);
+    static n: number;
+}
+C3.n;
+let c3 = new C3('hello');
+c3.message;
+
+declare let C4: typeof abstract class {
+    constructor(s: string);
+    static n: number;
+    s: string;
+}
+C4.n;
+let c4 = new C4('hello');  // Error
+
+declare let C5: typeof class<T> {
+    constructor(x: T);
+    x: T;
+}
+
+let c51 = new C5('hello');
+c51.x;
+let c52 = new C5(42);
+c52.x;
+
+type BoxFactory<T> = typeof class Box {
+    static default: T;
+    constructor(value?: T);
+    value: T;
+}
+
+declare let StringBox: BoxFactory<string>;
+StringBox.default;
+let sb = new StringBox('hello');
+sb.value;
+
+declare let NumberBox: BoxFactory<number>;
+NumberBox.default;
+let nb = new NumberBox(42);
+nb.value;
+
+declare const sb1: InstanceType<BoxFactory<string>>;
+sb1.value;
+
+declare const nb1: InstanceType<BoxFactory<number>>;
+nb1.value;
+
+function Printable1<T extends new (...args: any[]) => object>(Base: T) {
+    return class extends Base {
+        static foo: string;
+        print() {}
+    }
+}
+
+declare function Printable2<T extends new (...args: any[]) => object>(Base: T): typeof class extends Base {

```

Comment:
Is there a difference between `typeof class extends Base {` and `typeof class extends T {` here?

Reply:
Oh sorry, `T` is the constructor type, `Base` is the instance type.

Reply:
To be clear: `T` is the constructor _type_, `Base` is the constructor _value_.